bindings/python/Android.mk

```diff
@@ -74,15 +79,29 @@ LOCAL_GENERATED_SOURCES := $(generated-sources-dir)/pfw_wrap.cxx $(generated-sou
 
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(generated-sources-dir)
 
+# Careful, we need to invoque the android python config not the host's one.
+# BTW, the intenal install directory of python is hardcoded to a dummy value,
+# thus we need to manually add the correct path to libs to the library list.
+ifeq ($(HOST_OS), linux)
+LOCAL_LDLIBS += $(shell $(PYTHON_BIN_PATH)/python $(PYTHON_BIN_PATH)/python-config --ldflags) \
+                -L $(PYTHON_INSTALL_PATH)/lib/
+else
+LOCAL_LDLIBS += $(shell python-config --ldflags)
+endif
+
 $(generated-sources-dir)/pfw_wrap.h: $(generated-sources-dir)/pfw_wrap.cxx
 
 # The PyPfw.py file is generated in the directory given by -outdir switch, thus
 # this directory must be put in the python path to be reachable
+$(generated-sources-dir)/pfw_wrap.cxx: HOST_ARCH := $(HOST_ARCH)
+$(generated-sources-dir)/pfw_wrap.cxx: HOST_OS := $(HOST_OS)
+$(generated-sources-dir)/pfw_wrap.cxx: HOST_LIBRARY_PATH := $(HOST_LIBRARY_PATH)
```

Comment:
Not sure that this is needed as those variable should be immutable. Nevertheless, fool safe protection is not a bad idea in android build system.